include/onnxruntime/core/graph/graph.h

```diff
@@ -1423,23 +1423,33 @@ class Graph {
   // or a subgraph, so that the various operations that are part of the Resolve can work iteratively or
   // recursively as needed.
   struct ResolveContext {
-    ResolveContext() = default;
+    ResolveContext(const Graph& owning_graph) : graph{owning_graph} {
+    }
 
-    std::unordered_map<std::string, std::pair<Node*, int>> output_args;
-    std::unordered_set<std::string> inputs_and_initializers;
-    std::unordered_set<std::string> outer_scope_node_args;
-    std::unordered_map<std::string, NodeIndex> node_name_to_index;
+    std::unordered_map<std::string_view, std::pair<Node*, int>> output_args;
+    std::unordered_set<std::string_view> inputs_and_initializers;
+    std::unordered_map<std::string_view, NodeIndex> node_name_to_index;
```

Comment:
Since we are touching these already, any chance we want to use `InlinedHashMap`s ?

Reply:
IIRC the usage of those are small collections where we're trying to avoid a memory allocation by having a small stack allocated buffer. These collections will most likely be quite large so are very unlikely to be able to use the stack allocated buffer.